v0/destinations/hs/util.js

```diff
@@ -17,7 +133,168 @@ const getEmailAndUpdatedProps = properties => {
   };
 };
 
+/* NEW API util functions */
+
+/**
+ * look for the contact in hubspot and extract its contactId for updation
+ * @param {*} destination
+ * @returns
+ */
+const searchContacts = async (message, destination, lookupField = null) => {
+  const { Config } = destination;
+  let res;
+  let contactId = null;
+  const traits = getFieldValueFromMessage(message, "traits");
+  let propertyName;
+
+  if (lookupField) {
+    propertyName = lookupField;
+  } else {
+    propertyName = traits[`${Config.lookupField}`];
```

Comment:
- It seems like the payload will be
    
    ```jsx
    "traits": {
    "lookupFieldKey": "lookupFieldPropertyName",
    "lookupFieldPropertyName": "lookUpFieldValue"
    }
    
    ```
    
- why can’t we take `lookupFieldPropertyName` directly from the config and map the value from traits at once

Reply:
We do not want the events to be driven from config otherwise one needs to change the config every time for lookup field

Reply:
Is there any other way to handle a situation like this @utsabc @ItsSudip?

Reply:
It would same if you want this dynamic nature anyway you will have to use event payload